include/devicetree/pm_state.h

```diff
@@ -0,0 +1,165 @@
+/*
+ * Copyright (c) 2020 Intel corporation
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+#ifndef ZEPHYR_INCLUDE_DEVICETREE_PM_STATE_H_
+#define ZEPHYR_INCLUDE_DEVICETREE_PM_STATE_H_
+
+#ifdef __cplusplus
+extern "C" {
+#endif
+
+#include <power/power_state.h>
+
+/**
+ * @defgroup devicetree-pm_states Power Management states device tree macros
+ * @{
+ */
+
+/**
+ * @brief Construct a pm_state_info from 'pm_states' property at index 'i'
```

Comment:
I think @galak should confirm,  but I don't think it's right to place structure-creating macros in here.

I think instead the power_state.h header that defines the structure should provide a helper macro to create it from a devicetree node_id and other information as needed, similarly to DEVICE_DT_DEFINE.